builtin/providers/gitlab/resource_gitlab_project.go

```diff
@@ -82,6 +90,7 @@ func resourceGitlabProjectSetToState(d *schema.ResourceData, project *gitlab.Pro
 	d.Set("snippets_enabled", project.SnippetsEnabled)
 	d.Set("visibility_level", visibilityLevelToString(project.VisibilityLevel))
 
+	d.Set("id", project.ID)
```

Comment:
Sure, the docs are in the repo too.   Probably also worth adding the missing example to `gitlab_project_hooks` too, so the next user doesn't think that `"${gitlab_project.foo.id}"` isn't usable already.